query/src/sql/planner/plans/physical_scan.rs

```diff
@@ -12,39 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::any::Any;
+
 use crate::sql::optimizer::ColumnSet;
 use crate::sql::optimizer::PhysicalProperty;
-use crate::sql::optimizer::RequiredProperty;
 use crate::sql::optimizer::SExpr;
+use crate::sql::plans::BasePlan;
+use crate::sql::plans::LogicalPlan;
+use crate::sql::plans::PhysicalPlan;
+use crate::sql::plans::PlanType;
 use crate::sql::IndexType;
-use crate::sql::PhysicalPlan;
-use crate::sql::Plan;
 
-#[derive(Clone, Default, PartialEq, Debug)]
+#[derive(Clone, Debug)]
 pub struct PhysicalScan {
     pub table_index: IndexType,
     pub columns: ColumnSet,
 }
 
-impl PhysicalScan {
-    pub fn create(table_index: IndexType, columns: ColumnSet) -> Self {
-        PhysicalScan {
-            table_index,
-            columns,
-        }
+impl BasePlan for PhysicalScan {
+    fn plan_type(&self) -> PlanType {
+        PlanType::PhysicalScan
     }
-}
 
-impl PhysicalPlan for PhysicalScan {
-    fn compute_physical_prop(&self, _expression: &SExpr) -> PhysicalProperty {
-        PhysicalProperty::default()
+    fn is_physical(&self) -> bool {
```

Comment:
Is there a plan that both `physical` or `logical`？ Or neither `physical` and `logical`？

Reply:
Yes, `ProjectPlan` is both physical and logical.

And `PatternPlan` is neither physical nor logical.